### PR TITLE
adding support for VLANs (#192)

### LINKED ISSUE #192: support Linode VLANs
Linode has "VLAN" networks in addition to private/public v4/v6, allowing discovery of VLAN addresses would be useful

### PR DESCRIPTION
**Pull request by jriddle-linode**
updating tests to auto-generate instances
updating linodego to v1.8.1 to make use of new features

This adds the ability to search for VLANs by adding a vlan_label field to the request, which resolves #192

### CODE REVIEW

File: provider/linode/linode_discover_test.go [hunk 1]
```diff
@@ -114,3 +163,87 @@ func TestAddrsTaggedRegion(t *testing.T) {
...
+			t.Errorf("Error deleting test Instance: %s", terr)
+		}
+	}
+
+	return instance, teardown, nil
+}
+
+func getLinodeClient(t *testing.T) *linodego.Client {
+	t.Helper()
+	apiToken := os.Getenv("LINODE_API_TOKEN")
```
Comment (LBGarber) [inline provider/linode/linode_discover_test.go:237]: Could we make this `LINODE_TOKEN` to fall in line with our other integrations?

**Comment by jriddle-linode** [comment]
bump

**Comment by jriddle-linode** [comment]
tests pass locally but it looks like circle-ci needs updated to go1.18 for it to work.

**Comment by kaspergrubbe** [comment]
@LBGarber Can we do something here? This would be really helpful for those of us with clusters on Linode.

**Pull request closed without merging**